etcdserver/apply.go

```diff
@@ -59,7 +59,7 @@ type applierV3 interface {
 
 	AuthEnable() (*pb.AuthEnableResponse, error)
 	AuthDisable() (*pb.AuthDisableResponse, error)
-	Authenticate(r *pb.AuthenticateRequest) (*pb.AuthenticateResponse, error)
+	Authenticate(index uint64, r *pb.InternalAuthenticateRequest) (*pb.AuthenticateResponse, error)
```

Comment:
On the second thought, I think current interface would be enough simple and the context based interface would increase amount of code in unit tests (and change of interface cannot be detected as a type error). How do you think?